dmutils/documents.py

```diff
@@ -132,56 +134,93 @@ def upload_service_documents(uploader, upload_type, documents_url, service, requ
     return files, errors
 
 
-# some of the more... seemingly oddly specific functions here are apparently referenced by validators, hence their
-# existence
-
-
 def file_is_not_empty(file_contents):
     return not file_is_empty(file_contents)
 
 
-def file_is_empty(file_contents):
-    empty = len(file_contents.read(1)) == 0
-    file_contents.seek(0)
-    return empty
+def file_is_empty(file_object):
+    return len(get_first_128_bytes(file_object)) == 0
 
 
 def file_is_less_than_5mb(file_):
-    return get_file_size(file_) < FILE_SIZE_LIMIT
+    return get_file_size(file_) < 5400000
+
+
+def get_first_128_bytes(file_object):
+    """Get the contents of a file like object but replace the pointer when we're finished"""
+    position = file_object.tell()
+    file_object.seek(0)
+    contents = file_object.read()
+    file_object.seek(position)
+    return contents
+
+
+def get_extension(filename):
+    file_name, file_extension = os.path.splitext(filename)
+    return file_extension.lower()
+
+
+def get_possible_extensions_from_format(file_object):
+    """
+    Pass the first 128 bytes from the file object to fleep for a list of possible extensions
+
+    We don't need to load the whole file into memory to let fleep determine the file type. Apparently the first 128
+    bytes are sufficient: https://github.com/floyernick/fleep-py/blob/994bc2c274482d80ab13d89d8f7343eb316d3e44/
+        fleep/__init__.py#L55
+
+    :param file_object: File object open as bytes.
+    :return: list(str): Possible file formats.
+    """
+    return fleep.get(get_first_128_bytes(file_object)).extension
+
+
+def extension_matches_possible_file_formats(file_object, extension):
+    """
+    Pass the first 128 bytes from the file object to fleep and the supplied extension.
+
+    :param file_object: File object open as bytes.
+    :param extension: File object open as bytes.
+    :return: boolean: Does extension match guess at format?
+    """
+    return fleep.get(get_first_128_bytes(file_object)).extension_matches(extension.strip('.'))
 
 
 def file_is_open_document_format(file_object):
-    return get_extension(file_object.filename) in [
-        ".pdf", ".pda", ".odt", ".ods", ".odp"
```

Comment:
Does anyone use `.pda` docs? I know it's not in the fleek list, but should we allow `.pda` extensions without the format check?

Reply:
There's no such thing as far as I've been able to work out. Not in terms of a type of open doc. I brought it up on slack a while ago and no one had heard of it and said :partytrashfire: